crates/bevy_reflect/derive/src/impls/common.rs

```diff
@@ -156,3 +156,33 @@ pub fn common_partial_reflect_methods(
         #debug_fn
     }
 }
+
+pub fn reflect_auto_registration(meta: &ReflectMeta) -> Option<proc_macro2::TokenStream> {
+    if meta.attrs().no_auto_register() {
+        return None;
+    }
+
+    let bevy_reflect_path = meta.bevy_reflect_path();
+    let type_path = meta.type_path();
+    let generics = meta.type_path().generics();
+
+    if !generics.into_token_stream().is_empty() {
+        return None;
+    };
+
+    Some(quote! {
+        #[cfg(target_family = "wasm")]
+        #bevy_reflect_path::__macro_exports::wasm_init::wasm_init!{
+            #bevy_reflect_path::__macro_exports::AUTOMATIC_REFLECT_REGISTRATIONS
+                .write()
+                .expect("Failed to get write lock for automatic reflect type registration")
+                .push(<#type_path as #bevy_reflect_path::__macro_exports::RegisterForReflection>::__register);
+        }
```

Comment:
Seeing the differences between `inventory` and `wasm_init` I wonder if it makes sense to write an `inventory`-like wrapper on top of `wasm_init` (maybe as a separate crate?).

Reply:
I thought about it as well, but couldn't come up with a good enough design to merge the two. `inventory` requires a type to `collect!` and a const func for `submit!`, while `wasm_init` is needs to store stuff in a static and submit by calling a function. Every time I tried to create an abstraction over them it turned out even worse.

Reply:
Tried to somewhat abstract the platform differences, but since both macros are used in slightly different contexts, there is no guarantee that anything supported by one is going to be supported by the other. `inventory` is more restrictive and will be the one developers will most likely see when testing stuff, so maybe it's fine?

Reply:
It still feels a bit iffy but I would say it's ok for now. Feel free to resolve the comment.